src/System.Data.Common/src/System/Xml/XPathNodePointer.cs

```diff
@@ -1071,7 +1070,7 @@ private XmlNodeOrder CompareNamespacePosition(XPathNodePointer other)
             XPathNodePointer xp2 = other.Clone((DataDocumentXPathNavigator)(other._owner.Target));
             while (xp1.MoveToNextNamespace(XPathNamespaceScope.All))
             {
-                if (xp1.IsSamePosition(other))
+                if (xp1.IsSamePosition(xp2))
```

Comment:
@roji, could you help confirm what the right thing is to do here?  Seems like the intent here was to use `xp2` instead of `other`, so I made that change.  But the alternative is to just delete `xp2` and its `other.Clone`... in which case that also begs the question why we would need to clone one input and not the other.

Reply:
I'm afraid I don't have any knowledge about this part of System.Data... This definitely seem like a bug, and passing `xp2` definitely seems like the intent, but I don't have any more knowledge about this than anyone else... Removing `xp2` altogether seems like it would be wrong.

Reply:
Ok, thanks.